codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -316,10 +332,10 @@ public Http2Stream setPriority(int parentStreamId, short weight, boolean exclusi
             // Already have a priority. Re-prioritize the stream.
             weight(weight);
 
-            if (newParent != parent() || exclusive) {
+            if (newParent != parent() || (exclusive && newParent.numChildren() != 1)) {
```

Comment:
@Scottmitch should this check be `!newParent.isExclusiveChild(this)`, where `isExclusiveChild` would return `true` if `numChildren == 1` and the child is the one provided?